docs/smoothquant.md

```diff
@@ -0,0 +1,337 @@
+
+<div align="center">
+    <img src="./imgs/sq_pc.png"/>
+</div>
+
+
+## SmoothQuant and Our Enhancement
+
+### SmoothQuant
+
+In the previous subsection, we have explained why per-channel quantization could not be applied for activation, even though it could lead to lower quantization loss. However, the quantization error loss of activation plays an important role in the accuracy loss of model quantization[^2][^3][^4]. 
+
+
+
+To reduce the quantization loss of activations, lots of methods have been proposed. In the following, we briefly introduce SPIQ[^2], Outlier Suppression[^3] and Smoothquant[^4]. All these three methods share a similar idea to migrate the difficulty from activation quantization to weight quantization but differ in how much difficulty to be transferred.
+
+
+So **the first question is how to migrate the difficulty from activation to weights?** The solution is straightforward, that is to convert the network to an output equivalent network that is presented in the image below and apply quantization to this equivalent network. The intuition is that each channel of activation could be scaled to make it more quantization-friendly, similar to a fake per-channel activation quantization.
+
+<div align="center">
+    <img src="./imgs/sq_convert.png"/>
+</div>
+
+
+Please note that this conversion will make the quantization of weights more difficult, because the scales attached to weights shown above are per-input-channel, while quantization of weights is per-output-channel or per-tensor.
+
+So **the second question is how much difficulty to be migrated**, that is how to choose the **conversion per-channel scale** $s_{x1}$ and $s_{x2}$ from the above image. Different works adopt different ways.
+
+*SPIQ* just adopts the quantization scale of activations as the conversion per-channel scale.
+
+*Outlier suppression* adopts the scale of the preceding layernorm as the conversion per-channel scale.
+
+*Smoothquant* introduces a hyperparameter $\alpha$ as a smooth factor to calculate the conversion per-channel scale and balance the quantization difficulty of activation and weight.
+
+$$
+s_j = max(|X_j|)^\alpha/max(|W_j|)^{1-\alpha} \tag{4}
+$$
+
+j is the index of the input channels.
+
+
+
+<div align="center">
+    <img src="./imgs/smoothquant.png" height="250"/>
+</div>
+
+
+
+For most of the models such as OPT and BLOOM, $\alpha = 0.5$ is a well-balanced value to split the difficulty of weight and activation quantization. A larger $\alpha$ value could be used on models with more significant activation outliers to migrate more quantization difficulty to weights.
+
+
+### Our enhancement: 
+
+#### Algorithm: Layer-wise Auto-tuning of $\alpha$.

```

Comment:
$\ ?

Reply:
$\ is the latex format, $\alpha$  will show on README.